src/parser.rs

```diff
@@ -2156,6 +2155,7 @@ impl<'input, 'abbrev, 'unit, Endian> EntriesCursor<'input, 'abbrev, 'unit, Endia
             return Ok(None);
         }
 
+        let offset = self.current_offset();
```

Comment:
Sorry, I just realized I my previous comment wasn't complete. It also needs to calculate the offset using the value that `after_entry()` returns.